include/nbl/builtin/hlsl/glsl_compat/core.hlsl

```diff
@@ -245,8 +245,8 @@ struct equal_helper;
 #ifdef __HLSL_VERSION
 
 template<typename Vectorial>
-NBL_PARTIAL_REQ_TOP(concepts::Vectorial<Vectorial> && concepts::Integral<Vectorial>)
-struct equal_helper<Vectorial NBL_PARTIAL_REQ_BOT(concepts::Vectorial<Vectorial> && concepts::Integral<Vectorial>) >
+NBL_PARTIAL_REQ_TOP(always_true<decltype(spirv::IEqual<Vectorial>(experimental::declval<Vectorial>(),experimental::declval<Vectorial>()))> && concepts::Vectorial<Vectorial> && concepts::Integral<Vectorial>)
+struct equal_helper<Vectorial NBL_PARTIAL_REQ_BOT(always_true<decltype(spirv::IEqual<Vectorial>(experimental::declval<Vectorial>(),experimental::declval<Vectorial>()))> && concepts::Vectorial<Vectorial> && concepts::Integral<Vectorial>) >
```

Comment:
to avoid clutter can't you make a `spirv::IEqualISCallable` ? same for the `FOrdEqual`